File: packages/backend/internal/model/task.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Task struct {
	ID          bson.ObjectID `bson:"_id,omitempty"  json:"id"`
	UserID      bson.ObjectID `bson:"user_id"        json:"user_id"`
	Title       string             `bson:"title"          json:"title"`
	Description string             `bson:"description"    json:"description"`
	Status      string             `bson:"status"         json:"status"`   // "todo"|"in_progress"|"done"
	Priority    string             `bson:"priority"       json:"priority"` // "low"|"important"|"urgent"|"critical"|"routine"
	DueAt       *time.Time         `bson:"due_at"         json:"due_at"`
	IsDeleted   bool               `bson:"is_deleted"     json:"-"`
	DeletedAt   *time.Time         `bson:"deleted_at"     json:"-"`
	CreatedAt   time.Time          `bson:"created_at"     json:"created_at"`
	UpdatedAt   time.Time          `bson:"updated_at"     json:"updated_at"`
}

File: packages/frontend/src/api/task.ts
import request from "@/utils/request";

export interface Task {
  id: string;
  user_id: string;
  title: string;
  description: string;
  status: "todo" | "in_progress" | "done";
  priority: "low" | "important" | "urgent" | "critical" | "routine";
  due_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface ListTasksReq {
  status?: string;
  priority?: string;
  limit?: number;
  cursor?: string;
}

export interface ListTasksResp {
  items: Task[];
  next_cursor: string;
}

export interface CreateTaskReq {
  title: string;
  description?: string;
  priority?: string;
  due_at?: string;
}

export interface UpdateTaskReq {
  title?: string;
  status?: string;
  priority?: string;
  due_at?: string;
  description?: string;
}

export const taskApi = {
  getTasks(params?: ListTasksReq): Promise<ListTasksResp> {
    return request.get("/tasks", {
      params,
    }) as unknown as Promise<ListTasksResp>;
  },

  getTaskById(id: string): Promise<Task> {
    return request.get(`/tasks/${id}`) as unknown as Promise<Task>;
  },

  createTask(data: CreateTaskReq): Promise<Task> {
    return request.post("/tasks", data) as unknown as Promise<Task>;
  },

  updateTask(id: string, data: UpdateTaskReq): Promise<void> {
    return request.patch(`/tasks/${id}`, data) as unknown as Promise<void>;
  },

  deleteTask(id: string): Promise<void> {
    return request.delete(`/tasks/${id}`) as unknown as Promise<void>;
  },
};
